src/localization/database/cost_matrix.h

```diff
@@ -0,0 +1,45 @@
+#ifndef SRC_DATABASE_COST_MATRIX_H_
+#define SRC_DATABASE_COST_MATRIX_H_
+
+#include "features/feature_factory.h"
+
+#include <string>
+#include <vector>
+
+using Matrix = std::vector<std::vector<double>>;
+
+/**
+ * @brief  Stores costs as matrix.
+ */
+class CostMatrix {
+public:
+  CostMatrix(const std::string &costMatrixFile);
```

Comment:
A single argument constructor should be `explicit`. See https://google.github.io/styleguide/cppguide.html#Implicit_Conversions and https://youtu.be/TFoav6vhgdg